home_works/lesson14/handlers.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+
+class BaseHandler:
+    def __init__(self, path):
+        self.path = path
+        self._file = None
+
+    def read(self):
+        raise NotImplementedError()
+
+    def append(self, data):
+        raise NotImplementedError()
+
+    def close(self):
+        raise NotImplementedError()
+
+
+class JsonHandler(BaseHandler):
+    def read(self):
+        with open(self.path) as file:
+            content = file.read()
+        return content
+
+    def append(self, data):
+        try:
+            with open(self.path) as file:
+                self._file = json.load(file)
+        except (json.JSONDecodeError, ValueError):
+            self._file = []
+
+        self._file.append(data)
```

Comment:
hmm .. I think you should check that our data - is list

Reply:
I don't quite understand the conditions. If not the list, throw an error ?